lib/using_restricted_array.rb

```diff
@@ -6,56 +6,105 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Time complexity is O(1) because the arry has a fixed length of 20
+# Space complexity: I think also O(1) since it is a fixed amount that it can hold
 def length(array)
```

Comment:
👍 Although the time complexity is O(n) because you have to travel the length of the array and it will vary by array size.